src/de.rs

```diff
@@ -947,6 +969,16 @@ where
         }
         *self.len -= 1;
 
+        match self.de.peek()? {
+            Some(_byte @ 0x00..=0x1b) if !self.accept_packed => {
+                return Err(self.de.error(ErrorCode::WrongStructFormat));
+            }
+            Some(_byte @ 0x40..=0x7f) if !self.accept_named => {
```

Comment:
I used range 0x40 which is for byte string here. I felt I needed to mention it.

Reply:
I'm a bit confused, I thought fields would only use UTF-8 names and not binary names?

Reply:
ha, missed that one! well I'm not too sure. That's one of the reasons I raised this question.